sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_differentiate_custom_forms_with_labeled_and_unlabeled_models_async.py

```diff
@@ -0,0 +1,97 @@
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: sample_differentiate_custom_forms_with_labeled_and_unlabeled_models_async.py
+
+DESCRIPTION:
+    This sample demonstrates the differences in output that arise when recognize_custom_forms
+    is called with custom models trained with labeled and unlabeled data.
+USAGE:
+    python sample_differentiate_custom_forms_with_labeled_and_unlabeled_models_async.py
+
+    Set the environment variables with your own values before running the sample:
+    1) AZURE_FORM_RECOGNIZER_ENDPOINT - the endpoint to your Cognitive Services resource.
+    2) AZURE_FORM_RECOGNIZER_KEY - your Form Recognizer subscription key
+    3) CUSTOM_TRAINED_LABELED_MODEL_ID - the ID of your custom model trained with labeled data
+    4) CUSTOM_TRAINED_UNLABELED_MODEL_ID - the ID of your custom model trained with unlabeled data
+"""
+
+import os
+import asyncio
+
+
+class DifferentiateCustomFormsWithLabeledAndUnlabeledModelsAsync(object):
+
+    endpoint = os.environ["AZURE_FORM_RECOGNIZER_ENDPOINT"]
+    key = os.environ["AZURE_FORM_RECOGNIZER_KEY"]
+    labeled_model_id = os.environ["CUSTOM_TRAINED_LABELED_MODEL_ID"]
+    unlabeled_model_id = os.environ["CUSTOM_TRAINED_UNLABELED_MODEL_ID"]
+
+    async def recognize_custom_forms(self):
+        from azure.core.credentials import AzureKeyCredential
+        from azure.ai.formrecognizer.aio import FormRecognizerClient
+        async with FormRecognizerClient(
+            endpoint=self.endpoint, credential=AzureKeyCredential(self.key)
+        ) as form_recognizer_client:
+
+            # The form you are recognizing must be of the same type as the forms the custom model was trained on
+            with open("sample_forms/forms/Form_1.jpg", "rb") as f:
+                stream = f.read()
+            forms_with_labeled_model = await form_recognizer_client.recognize_custom_forms(
+                model_id=self.labeled_model_id, stream=stream
+            )
+            forms_with_unlabeled_model = await form_recognizer_client.recognize_custom_forms(
+                model_id=self.unlabeled_model_id, stream=stream
+            )
+
+            # The main difference is found in the labels of its fields
+            # The form recognized with a labeled model will have the labels it was trained with,
+            # the unlabeled one will be denoted with indices
+            print("--------Recognizing forms with labeled custom model--------")
+            for labeled_form in forms_with_labeled_model:
+                for label, field in labeled_form.fields.items():
+                    # With your labeled custom model, you will not get back label data but will get back value data
+                    # This is because your custom model didn't have to use any machine learning to deduce the label,
+                    # the label was directly provided to it
```

Comment:
It'd be better to remove this comment - the training time label is found in the name field, and it's better to keep the option open if the supervised team wanted to start populating the form field labels at some point in the future.

Reply:
I think I'll keep it for now because that is a notable difference between labeled and unlabeled at this point. Will remove if they end up populating label data in the future

Reply:
It would be good to add that the training-time label is returned in the name field.